docs/rules/no-type-literal.md

```diff
@@ -0,0 +1,121 @@
+# Enforces the use of interfaces over type literals (`type T = {...}`).
+
+In TypeScript, type literals can act sort of like interfaces, but there are subtle differences that 
+make interfaces a better choice:   
+- Interfaces can be extended or implemented by other types.
+- Interfaces create a new name that is used everywhere, so when debugging is easier to identify 
+the base type of an object.
+
+## Rule Details
+
+This rule aims to standardise the use of interfaces across the codebase.
+
+## Options
```

Comment:
As with the other PR, please can we stick to the exceptions being `false` by default for "no-" rules, I feel it is much clearer to users of the rule.